src/sources/soundsourceffmpeg.cpp

```diff
@@ -1,69 +1,145 @@
+        } else if (!strcmp(l_SInputFmt->name, "ogg")) {
+            list.append("ogg");
+        } else if (!strcmp(l_SInputFmt->name, "mov,mp4,m4a,3gp,3g2,mj2")) {
+            list.append("m4a");
+        } else if (!strcmp(l_SInputFmt->name, "mp4")) {
+            list.append("mp4");
+        } else if (!strcmp(l_SInputFmt->name, "mp3")) {
+            list.append("mp3");
+        } else if (!strcmp(l_SInputFmt->name, "aac")) {
+            list.append("aac");
+        } else if (!strcmp(l_SInputFmt->name, "opus") ||
+                   !strcmp(l_SInputFmt->name, "libopus")) {
+            list.append("opus");
+        } else if (!strcmp(l_SInputFmt->name, "wma")) {
+            list.append("wma");
+        }
+    }
+
+    return list;
+}
+
+SoundSourceFFmpeg::SoundSourceFFmpeg(QUrl url)
+    : SoundSource(url),
+      m_pFormatCtx(NULL),
+      m_iAudioStream(-1),
+      m_pCodecCtx(NULL),
+      m_pCodec(NULL),
+      m_pResample(NULL),
+      m_iCurrentMixxTs(0),
+      m_bIsSeeked(false),
+      m_lCacheBytePos(0),
+      m_lCacheStartByte(0),
+      m_lCacheEndByte(0),
+      m_lCacheLastPos(0),
+      m_lLastStoredPos(0),
+      m_lStoredSeekPoint(-1) {
 }
 
 SoundSourceFFmpeg::~SoundSourceFFmpeg() {
-    struct ffmpegLocationObject *l_SRmJmp = NULL;
+    close();
+}
+
+Result SoundSourceFFmpeg::tryOpen(SINT channelCountHint) {
+    unsigned int i;
+    AVDictionary *l_iFormatOpts = NULL;
+
+    const QByteArray qBAFilename(getLocalFileNameBytes());
+    qDebug() << "New SoundSourceFFmpeg :" << qBAFilename << "(channelCountHint:" << channelCountHint << ")";
+
+    DEBUG_ASSERT(!m_pFormatCtx);
+    m_pFormatCtx = avformat_alloc_context();
+
+#if LIBAVCODEC_VERSION_INT < 3622144
+    m_pFormatCtx->max_analyze_duration = 999999999;
+#endif
+
+    // Open file and make m_pFormatCtx
+    if (avformat_open_input(&m_pFormatCtx, qBAFilename.constData(), NULL,
+                            &l_iFormatOpts)!=0) {
+        qDebug() << "av_open_input_file: cannot open" << qBAFilename;
+        return ERR;
+    }
+
+#if LIBAVCODEC_VERSION_INT > 3544932
+    av_dict_free(&l_iFormatOpts);
+#endif
+
+    // Retrieve stream information
+    if (avformat_find_stream_info(m_pFormatCtx, NULL)<0) {
+        qDebug() << "av_find_stream_info: cannot open" << qBAFilename;
+        return ERR;
+    }
+
+    //debug only (Enable if needed)
+    //av_dump_format(m_pFormatCtx, 0, qBAFilename.constData(), false);
+
+    // Find the first audio stream
+    m_iAudioStream=-1;
+
+    for (i=0; i<m_pFormatCtx->nb_streams; i++)
+        if (m_pFormatCtx->streams[i]->codec->codec_type==AVMEDIA_TYPE_AUDIO) {
+            m_iAudioStream=i;
+            break;
+        }
+    if (m_iAudioStream==-1) {
+        qDebug() << "ffmpeg: cannot find an audio stream: cannot open"
+                 << qBAFilename;
+        return ERR;
+    }
+
+    // Get a pointer to the codec context for the audio stream
+    m_pCodecCtx=m_pFormatCtx->streams[m_iAudioStream]->codec;
```

Comment:
In this file we have missing spaces around some operators like " == "
If one has fun to fix it .. else never mind.